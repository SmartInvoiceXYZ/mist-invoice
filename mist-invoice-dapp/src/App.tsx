import {
  ChakraProvider,
  ColorModeScript,
  CSSReset
} from "@chakra-ui/react";
import theme from "@chakra-ui/theme";
import { Global } from "@emotion/react";
import { ErrorBoundary } from "react-error-boundary";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import "./App.css";
import { ErrorHandler } from "./components/ErrorHandler";
import { Web3ContextProvider } from "./context";
import Home from "./pages/Home";
import SelectInvoiceType from "./pages/SelectInvoiceType";
import { Layout } from "./shared/Layout";
import { globalStyles } from "./theme";

const App = () => (
  <ChakraProvider theme={theme}>
    <ColorModeScript initialColorMode={theme.config.initialColorMode} />
    <CSSReset />
    <Global styles={globalStyles} />
    <ErrorBoundary FallbackComponent={ErrorHandler}>
      <Web3ContextProvider>
        <BrowserRouter>
          <Layout>
            <Routes>
              <Route path="/" element={<Home />} />
              <Route path="/create" element={<SelectInvoiceType />} />
            </Routes>
          </Layout>
        </BrowserRouter>
      </Web3ContextProvider>
    </ErrorBoundary>
  </ChakraProvider>
);


export default App;
