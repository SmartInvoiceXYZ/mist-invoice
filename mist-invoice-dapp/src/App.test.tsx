import { render } from "@testing-library/react";
import React from "react";
import App from "./App";

jest.mock("./pages/Home", () => () => <div>Home</div>);
jest.mock("./pages/SelectInvoiceType", () => () => (
  <div>SelectInvoiceType</div>
));
jest.mock("./shared/Layout", () => (children: React.ReactElement) => <div test-id="div:layout">{children}</div>);

it("App renders", () => {
  const view = render(<App />);

  expect(view.asFragment()).toMatchSnapshot();
});
