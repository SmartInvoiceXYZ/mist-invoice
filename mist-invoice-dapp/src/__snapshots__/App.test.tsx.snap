// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<DocumentFragment>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="light",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <div
    class="css-149uwkc"
  >
    <div
      class="css-gh38gr"
    >
      <div
        class="css-1hskriy"
      >
        <a
          href="/invoices"
        >
          <div
            class="css-11dqbsu"
          >
            <img
              alt="Smart Invoice"
              class="chakra-image css-12aolnz"
              src="normal.svg"
            />
          </div>
        </a>
      </div>
      <div
        class="css-16fbqg3"
      >
        <a
          class="chakra-link css-spn4bz"
          href="/invoices"
        >
          Dashboard
        </a>
        <a
          class="chakra-link css-spn4bz"
          href="https://docs.smartinvoice.xyz"
          rel="noopener"
          target="_blank"
        >
          Documentation
        </a>
        <a
          class="chakra-link css-spn4bz"
          href="https://docs.smartinvoice.xyz/misc/get-support"
          rel="noopener"
          target="_blank"
        >
          Support
        </a>
      </div>
      <div
        class="css-uqd54k"
      />
      <div
        class="css-1x6ahq1"
      >
        <button
          class="chakra-button css-qoa0t0"
          type="button"
        >
          Dashboard
        </button>
        <a
          class="chakra-link css-1y6zyml"
          href="https://docs.smartinvoice.xyz"
          rel="noopener"
          target="_blank"
        >
          <span
            class="chakra-button css-qoa0t0"
          >
            Documentation
          </span>
        </a>
        <a
          class="chakra-link css-1y6zyml"
          href="https://docs.smartinvoice.xyz/misc/get-support"
          rel="noopener"
          target="_blank"
        >
          <span
            class="chakra-button css-qoa0t0"
          >
            Support
          </span>
        </a>
      </div>
    </div>
    <div
      class="css-90ijcl"
    >
      <div>
        Home
      </div>
    </div>
    <div
      class="css-15cs0q7"
    >
      <div
        class="css-vl5bay"
      >
        <a
          class="chakra-link css-spn4bz"
          href="/invoices"
        >
          <img
            alt="Smart Invoice"
            class="chakra-image css-lu3yos"
            src="white.svg"
          />
        </a>
        <div
          class="css-16fbqg3"
        >
          <a
            class="chakra-link css-spn4bz"
            href="https://smartinvoice.xyz"
            rel="noopener"
            target="_blank"
          >
            Website
          </a>
          <a
            class="chakra-link css-spn4bz"
            href="https://docs.smartinvoice.xyz"
            rel="noopener"
            target="_blank"
          >
            Documentation
          </a>
          <a
            class="chakra-link css-spn4bz"
            href="https://docs.smartinvoice.xyz/misc/get-support"
            rel="noopener"
            target="_blank"
          >
            Support
          </a>
          <a
            class="chakra-link css-spn4bz"
            href="https://twitter.com/SmartInvoiceXYZ"
            rel="noopener"
            target="_blank"
          >
            Twitter
          </a>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
